### ISSUE TITLE
Localization support

### ISSUE THREAD

**Issue by fedebaier** [stripe/stripe-react-native#243]
I'm implementing the **Pre-built payments UI** and couldn't find a way to set a locale to show it in different languages. Is there something I'm missing or is this not yet implemented?

It's quite important for our app, since it supports a few languages.

**Comment by thorsten-stripe** [comment]
@fedebaier thanks for the feedback! Currently the sheet is localised based on the language in the phone device settings. But I agree that we should allow you to overwrite it. I've forwarded this to the team.

**Comment by fedebaier** [comment]
I've tested changing the device's language but it always shows in English.

**Comment by thorsten-stripe** [comment]
If you go to Settings -> System -> Languages & Input -> Languages -> and then add your language to the top of the list, you should see a different version when you go back to the payment sheet.

@fedebaier have you tested both platforms?

**Comment by fedebaier** [comment]
Yes, I've tried this in both iOS and Android, on simulators and on physical devices, and it always displays in English.

**Comment by thorsten-stripe** [comment]
@fedebaier which languages have you set for your apps? It's working fine for me. Do note that on iOS you need to specify which languages your app supports: https://help.apple.com/xcode/mac/current/#/devafa3a605f on Android that isn't required.

* Android
![image](https://user-images.githubusercontent.com/23213994/119848074-19672600-bf3e-11eb-9a5a-cc532fee1dbc.png)

* iOS
![image](https://user-images.githubusercontent.com/23213994/119848005-048a9280-bf3e-11eb-8604-10c8800422bc.png)

**Comment by fedebaier** [comment]
It's working now on Android (maybe I didn't test properly there), but the issue persists on iOS, even after specifying the languages like you suggested:

<img width="322" alt="Screenshot 2021-05-27 at 17 20 46" src="https://user-images.githubusercontent.com/2706853/119853492-88c62100-bf10-11eb-93ae-45d2a662b36b.png">

![Simulator Screen Shot - iPhone 11 Pro Max - 2021-05-27 at 17 18 51](https://user-images.githubusercontent.com/2706853/119853508-8ebc0200-bf10-11eb-8bf2-fcd673042b8b.png)

![Simulator Screen Shot - iPhone 11 Pro Max - 2021-05-27 at 17 19 16](https://user-images.githubusercontent.com/2706853/119853528-924f8900-bf10-11eb-9438-975b4f845af2.png)

**Comment by harbolaez** [comment]
Same here, after adding Spanish through Xcode and building. The issue seems to persist in iOS.

**Comment by harbolaez** [comment]
fixed mine, by adding to info.plist: 
```
<key>CFBundleAllowMixedLocalizations</key>
  <string>true</string>
<key>CFBundleLocalizations</key>
<array>
    <string>en</string>
    <string>es</string>
</array>
```

**Comment by fedebaier** [comment]
> fixed mine, by adding to info.plist:
> 
> ```
> <key>CFBundleAllowMixedLocalizations</key>
> <string>true</string>
> <key>CFBundleLocalizations</key>
> <array>
>     <string>en</string>
>     <string>es</string>
> </array>
> ```

That solved the issue also for me, thanks! Still would be nice to have the option to pass a locale to the payment sheet.

**Comment by SlavisDEV** [comment]
> 
> 
> fixed mine, by adding to info.plist:
> 
> ```
> <key>CFBundleAllowMixedLocalizations</key>
> <string>true</string>
> <key>CFBundleLocalizations</key>
> <array>
>     <string>en</string>
>     <string>es</string>
> </array>
> ```

This does not work for me. Anyone has similiar issue? I tried to set language directly inside Info.plist and by Xcode -> General as well.

**Comment by nmehlei** [comment]
@thorsten-stripe Is it planned to support an override for the language? If yes, is there an estimation when this will become available? In my case, the app I'm developing should display in a different language than my device's language.

**Comment by angelorobo** [comment]
@nmehlei Have you found a solution yet? We also want to support to change locale of the payment form.

**Comment by thomasyim02** [comment]
+1 This is an enhancement we are waiting for~

**Comment by charliecruzan-stripe** [comment]
Does editing info.plist not work for folks that are commenting here? You should be able to programmatically change the app locale, which would be a good solution for @nmehlei's issue (I think)

```
<key>CFBundleAllowMixedLocalizations</key>
<string>true</string>
<key>CFBundleLocalizations</key>
<array>
    <string>en</string>
    <string>es</string>
</array>
```

**Comment by thomasyim02** [comment]
@charliecruzan-stripe 
Do you mean we programmatically change the app locale in the native side of the app?

**Comment by charliecruzan-stripe** [comment]
Yes, I'm pretty sure there are methods to do that on both iOS and Android (although it may take some digging, and I'm not sure if it's generally recommended to override the user's locale)

**Comment by AbdullahJaved-dev** [comment]
@thorsten-stripe I want to change the payment sheet language with InApp locale for Andoid. Is there a way to do it?
![241e6744-12fd-448a-9455-f5df888aa388](https://user-images.githubusercontent.com/66158133/203346554-b5fd1b29-8e23-49f3-aa9b-2672d2b32000.jpg)

**Comment by charliecruzan-stripe** [comment]
The payment sheet localizes itself to use the device's current language settings. What's the use case for overriding the user's own language setting?

**Comment by AbdullahJaved-dev** [comment]
> The payment sheet localizes itself to use the device's current language settings. What's the use case for overriding the user's own language setting?

With device language change payment sheet works fine. But I want to change app language without changing device language. In this scenario payment sheet is not translated.

**Comment by DavitVosk** [comment]
Do we have any progress for this please @charliecruzan-stripe @thorsten-stripe? Our app is supporting different languages and we would like the payment sheet to be localized in the same language as our app, but it listens to device language only 🧐

**Comment by angelica-snowit** [comment]
The language option should be at top level, not only the payment sheet, in order to localize also error messages.

**Comment by cufano** [comment]
Same here. Our app use multiple languages and the payment sheet should be localized in one of them dinamically.

**Comment by d4n1b** [comment]
Hi everyone, is there any update on this? - as @angelica-snowit mentioned, even though the Payment Sheet is showing the right language depending on the OS settings, I can still see the `localizedMessage` errors in English, i.e.:

```
{
  "stripeErrorCode": null,
  "declineCode": null,
  "localizedMessage": "The payment has been canceled",  <- HERE
  "message": "The payment has been canceled",
  "type": null,
  "code": "Canceled"
}
``` 

We are trying at least to provide our own translation to those messages, but unsure if it's safe to rely on the `code`: "Canceled".

Thanks!

**Comment by kothihaaung** [comment]
I created payment sheet with custom string to solve this issue.
https://github.com/kothihaaung/stripe-paymentsheet-customlabel
![Uploading screenshot_thai.png…]()

**Comment by ahmadkhalaf1** [comment]
what is the current status of this? i would like to change the language without changing the phone language , also react native 0.73 and android 14 , language change wont change the payment sheet for me.

**Comment by ArnaudFeelbat** [comment]
@ahmadkhalaf1 did you find any solution for your specific case ? Im facing the same issue

**Comment by aliwaqar981** [comment]
Anyone working on this?

**Comment by drilko** [comment]
Hi, is there any news regarding this issue?

**Comment by RobertMrowiec** [comment]
Hey hey, really no update on that? That sucks Stripe team. In newer RN versions Info.plist doesnt have 
'CFBundleLocalizations', nor xcode language list update helps... How should we manipulate the language in the payment sheet? It's always english

**Comment by AndrianarivoAlpha** [comment]
just add locale = "language you want" in StripeProvider like 
<StripeProvider
  publishableKey={""}
  merchantIdentifier="merchant.identifier"
  urlScheme="your-url-scheme"
  locale="fr">
<You App/>
</StripeProvider>

**Comment by RobertMrowiec** [comment]
> just add locale = "language you want" in StripeProvider like <StripeProvider publishableKey={""} merchantIdentifier="merchant.identifier" urlScheme="your-url-scheme" locale="fr">

Well, where in documentation you have this information? I can't find it anywhere.

**Comment by krvinc0de** [comment]
Any solution?

**Comment by AndrianarivoAlpha** [comment]
> > just add locale = "language you want" in StripeProvider like <StripeProvider publishableKey={""} merchantIdentifier="merchant.identifier" urlScheme="your-url-scheme" locale="fr">
> 
> Well, where in documentation you have this information? I can't find it anywhere.

nowhere but it works on android, or it just a coincidence i don't know!

**Comment by aacoimbra** [comment]
Same problem here. Localization is working for Android but not for IOS.

**Comment by fkeloks** [comment]
I use Expo Go on a French iOS device. The Expo Go app is in English (I can't change it), Stripe's interfaces are in English. I also need to change the language used by Stripe but obviously we can't do it.

**Comment by nickschinestzki** [comment]
Hey, here to request this feature in 2025. That's clearly a real need on the community.

**Comment by Under-Warz** [comment]
Wow, no news from 2021 on this really basic feature...

**Comment by overlineink** [comment]
Is there a way to inform stripe a specific region without user interaction?

**Comment by martinezguillaume** [comment]
For anyone using expo !
For Android, Stripe sheets will take the device system locale but it not working like that in iOS.

It is not possible to choose dynamically the language but you can, at least, choose one language
Here is the expo plugin that I'm using (here i'm setting the locale to fr)

```tsx
import {withXcodeProject, ConfigPlugin} from 'expo/config-plugins'

// https://github.com/stripe/stripe-react-native/issues/243
const withFixStripeLocalization: ConfigPlugin = config => {
  return withXcodeProject(config, async config => {
    const xcodeProject = config.modResults
    // All methods are from this lib https://github.com/apache/cordova-node-xcode
    xcodeProject.addKnownRegion('fr')
    xcodeProject.removeKnownRegion('en')

    // https://github.com/apache/cordova-node-xcode/issues/122#issuecomment-1043549645
    xcodeProject.pbxProjectSection()[xcodeProject.getFirstProject()['uuid']][
      'developmentRegion'
    ] = 'fr'
    return config
  })
}

export default withFixStripeLocalization
```

**Comment by hchapel** [comment]
Would be great to have different language in IOS paymentSheet, make it happen soon Team Stripe pleasee ❤️

**Comment by omartin** [comment]
It's your oldest open issue, time to fix it Stripe team ?

**Comment by thomsala** [comment]
@omartin yes it would be great ahah

**Comment by marinadeotti** [comment]
Would be great to have it fixed before Christmas 2025.🎅🏻🎄
